### PR TITLE
[ot] Update ot types

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/codesandbox/codesandbox-client/blob/master/packages/%40types/ot/index.d.ts & https://github.com/Operational-Transformation/ot.js/blob/master/lib/selection.js
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## ot ([<kbd>unpkg</kbd>](https://unpkg.com/browse/ot@latest/))
was missing the following properties:
1. version
1. SimpleTextOperation
1. Client
1. Server
1. EditorSocketIOServer
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d2fd16b2931a7728613b4b4abe326b9bfcd321fc
</p>

File: types/ot/index.d.ts [hunk 1]
```diff
@@ -6,36 +6,47 @@
...
 
     static fromJSON(operation: SerializedTextOperation): TextOperation;
-    toJSON(): SerializedTextOperation;
+    static isDelete(operation: string | number): boolean;
+    static isInsert(operation: string | number): boolean;
+    static isRetain(operation: string | number): boolean;
+    static transform(left: TextOperation, right: TextOperation): TextOperation;
 }
+
+export const version: '0.0.15';
```
Comment (amcasey) [inline types/ot/index.d.ts:52]: Would `string` be more appropriate?  Doing it this way would seem to require that every version have a distinct .d.ts file, even if this is the only change.  And callers are probably consuming it as a string?

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/ot/index.d.ts (hunk 1 above)
Comment (MichaelDeBoey) [reply types/ot/index.d.ts:52]: @amcasy `ot` is last updated about 6 years ago, so I don't think it matters here.
It's an extra confirmation to know these types are about the latest version.

Comment (amcasey) [reply types/ot/index.d.ts:52]: On the flip side, what's the benefit of giving the constant a literal type?  Maybe it's to cause some old code to be flagged as unreachable?
```ts
if (version === "0.0.14") {
    // unreachable
}
```

Comment (MichaelDeBoey) [reply types/ot/index.d.ts:52]: Good point, I'll change it to just string 🙂

**Review by amcasey — APPROVED** [review]
Thanks!

**Comment by MichaelDeBoey** [comment]
Ready to merge

**Pull request merged**